File: src/main/java/com/oriaxx77/oaths/domain/enity/Oath.scala
package com.oriaxx77.oaths.domain.enity

import javax.persistence.Entity
import scala.beans.BeanProperty
import javax.persistence.GeneratedValue
import javax.persistence.GenerationType
import javax.persistence.Id
import java.util.Date
import javax.persistence.ManyToOne
import javax.persistence.FetchType

@Entity
class Oath(@BeanProperty var oath: String) {

  @Deprecated
  def this() { this("") }

  @ManyToOne(fetch=FetchType.LAZY, optional=false)
  @BeanProperty
  var oathTaker: User = _

  @Id
  @GeneratedValue(strategy = GenerationType.AUTO)
  var id: Long = _
  var temptationFailedCount: Long = 0
  var temptationOvercomeCount: Long = 0
  var createdAt: Date = new Date()
  var lastModified: Date = new Date()

  override def toString = "Oath{ id: " + id + ", oath:" + oath + "}"
}

File: src/main/java/com/oriaxx77/oaths/domain/enity/Oath.java
package com.oriaxx77.oaths.domain.enity;

import java.util.Date;

import javax.persistence.Entity;
import javax.persistence.FetchType;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.ManyToOne;
import javax.persistence.NamedAttributeNode;
import javax.persistence.NamedEntityGraph;
import javax.persistence.PrePersist;
import javax.persistence.PreUpdate;
import javax.persistence.Temporal;
import javax.persistence.TemporalType;

import lombok.Getter;
import lombok.Setter;
import lombok.ToString;

@Entity
@Getter
@Setter
@ToString
@NamedEntityGraph( name="Oath.oathTaker", attributeNodes=@NamedAttributeNode("oathTaker") )
public class Oath
{
	@Id
	@GeneratedValue(strategy = GenerationType.AUTO)
	private Long id;

	@ManyToOne( fetch=FetchType.LAZY, optional=false )
	private User oathTaker;
	
	private Long temptationFailedCount;
	
	private Long temptationOvercomeCount;
	
	@Temporal(TemporalType.TIMESTAMP)
	private Date created;
	
	@Temporal(TemporalType.TIMESTAMP)
	private Date updated;
	
	private String oath;

	public Oath(String oath)
	{
		this.oath = oath;
	}

	public Oath()
	{
		this("");
	}
	
	@PrePersist
	protected void onCreate() 
	{
		created = new Date();
	}

	@PreUpdate
	protected void onUpdate() 
	{
		updated = new Date();
	}
	
}